framework/src/play-jdbc/src/main/resources/reference.conf

```
@@ -98,7 +98,7 @@ play {
         # 3. A value less than zero will bypass any initial connection attempt, and the pool will start immediately
         #    while trying to obtain connections in the background. Consequently, later efforts to obtain a connection
         #    may fail.
-        initializationFailTimeout = 1
+        initializationFailTimeout = -1
```

So, this was cherry-picked too, but I wonder if we should keep the old value.

---

See the tests too: https://github.com/playframework/playframework/pull/8502/files#diff-9d5fa835745ced04a4110c62681c137bR76

---

I slightly prefer keeping the old value just for compatibility reasons, but I wonder if anyone is actually depending on the current default behavior.

---

btw. if somebody uses migrations, what will happen with them, when the database is offline, while the application starts?

---

Revert to use the fail fast configuration.

---

> btw. if somebody uses migrations, what will happen with them, when the database is offline, while the application starts?

`ApplicationEvolutions` is loaded eagerly and its behavior depends on environment mode. In prod, if configured to `autoApply`, it will try to `evolve` which requires a connection. If configured to fail fast, then it will try to obtain a connection and fail after the time configured at `connectionTimeout` (defaults to 30 seconds).

---

Which means that, if you have evolutions and your connection pool is configured to NOT fail fast, then the application initialization will fail after the time defined by `connectionTimeout`.